### PR TITLE
[mixch] add support for mixch.tv

### PR DESCRIPTION
**Pull request by nao20010128nao**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR adds support for mixch.tv which is a streaming service in Japan.

Live URL: https://mixch.tv/u/16236849/live
(you can find another one here: https://mixch.tv/)

upstream issue:
- ytdl-org/youtube-dl#28727

### CODE REVIEW

File: yt_dlp/extractor/mixch.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    traverse_obj,
+)
+
+
+class MixchIE(InfoExtractor):
+    IE_NAME = 'mixch'
+    # allow omitting last /live in the URL, though it's likely uncommon
```
Comment (pukkandan) [inline yt_dlp/extractor/mixch.py:12]:
```suggestion
```
doesnt matter as long as it doesnt conflict with any other extractor

File: yt_dlp/extractor/mixch.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    _VALID_URL = r'https?://(?:www\.)?mixch\.tv/u/(?P<id>\d+)'
+
+    TESTS = [{
+        'url': 'https://mixch.tv/u/16137876/live',
+        'skip': 'live has ended',
```
Comment (pukkandan) [inline yt_dlp/extractor/mixch.py:17]: either add fields or make it `only_matching`. Though idk why this test is even needed

File: yt_dlp/extractor/mixch.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            'id': video_id,
+            'title': traverse_obj(initial_js_state, ('liveInfo', 'title')),
+            'comment_count': traverse_obj(initial_js_state, ('liveInfo', 'comments')),
+            'view_count': traverse_obj(initial_js_state, ('liveInfo', 'visitor')),
+            'timestamp': traverse_obj(initial_js_state, ('liveInfo', 'created')),
+            'uploader': traverse_obj(initial_js_state, ('broadcasterInfo', 'name')),
+            'uploader_id': video_id,
+            'formats': formats,
+            'is_live': True,
+            'webpage_url': url,
```
Comment (pukkandan) [inline yt_dlp/extractor/mixch.py:57]: any particular reason to override the webpage_url?

File: yt_dlp/extractor/mixch.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        video_id = self._match_id(url)
+        url = 'https://mixch.tv/u/%s/live' % video_id
+        webpage = self._download_webpage(url, video_id)
+
+        initial_js_state = self._parse_json(self._search_regex(
+            r'(?m)^\s*window\.__INITIAL_JS_STATE__\s*=\s*(\{.+?\});\s*$', webpage, 'initial JS state'), video_id)
+        if not initial_js_state.get('liveInfo'):
+            raise ExtractorError('Live has ended.', expected=True)
+
+        # the service does not provide alternative resolutions
```
Comment (pukkandan) [inline yt_dlp/extractor/mixch.py:42]: ?

File: yt_dlp/extractor/mixch.py (hunk 2 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/mixch.py:17]: https://github.com/yt-dlp/yt-dlp/pull/866#discussion_r701060313

File: yt_dlp/extractor/mixch.py (hunk 3 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/mixch.py:56]: to complement `/live` in URL

File: yt_dlp/extractor/mixch.py (hunk 2 above)
Comment (pukkandan) [reply yt_dlp/extractor/mixch.py:16]: That's what the other test is for, no?

Comment (nao20010128nao) [reply yt_dlp/extractor/mixch.py]: oh
converted to only_matching test

File: yt_dlp/extractor/mixch.py (hunk 4 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/mixch.py]: simplified

**Review by pukkandan — APPROVED** [review]

**Pull request merged**